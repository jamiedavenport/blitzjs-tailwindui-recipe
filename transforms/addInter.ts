import j from "jscodeshift";
import { Collection } from "jscodeshift/src/Collection";

export default function transform(program: Collection<j.Program>) {
  program.findJSXElements("DocumentHead").replaceWith((nodePath) => {
    return j.jsxElement(
      j.jsxOpeningElement(j.jsxIdentifier("DocumentHead")),
      j.jsxClosingElement(j.jsxIdentifier("DocumentHead")),
      [
        j.jsxText("\n"),
        j.jsxElement(
          j.jsxOpeningElement(
            j.jsxIdentifier("link"),
            [
              j.jsxAttribute(j.jsxIdentifier("rel"), j.literal("stylesheet")),
              j.jsxAttribute(
                j.jsxIdentifier("href"),
                j.literal("https://rsms.me/inter/inter.css")
              ),
            ],
            true
          ),
          null,
          []
        ),
        j.jsxText("\n"),
      ]
    );
  });
  return program;
}
