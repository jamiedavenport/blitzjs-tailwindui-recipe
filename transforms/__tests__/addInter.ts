import j from "jscodeshift";
import addInter from "../addInter";

const sampleFile = `import {Document, Html, DocumentHead, Main, BlitzScript /*DocumentContext*/} from "@blitzjs/core"

class MyDocument extends Document {
  // Only uncomment if you need to customize this behaviour
  // static async getInitialProps(ctx: DocumentContext) {
  //   const initialProps = await Document.getInitialProps(ctx)
  //   return {...initialProps}
  // }

  render() {
    return (
      <Html lang="en">
        <DocumentHead />
        <body>
          <Main />
          <BlitzScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument`;

test("addInter transformation", () => {
  expect(addInter(j(sampleFile)).toSource()).toMatchSnapshot();
});
