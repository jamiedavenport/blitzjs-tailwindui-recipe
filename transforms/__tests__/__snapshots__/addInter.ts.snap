// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addInter transformation 1`] = `
"import {Document, Html, DocumentHead, Main, BlitzScript /*DocumentContext*/} from \\"@blitzjs/core\\"

class MyDocument extends Document {
  // Only uncomment if you need to customize this behaviour
  // static async getInitialProps(ctx: DocumentContext) {
  //   const initialProps = await Document.getInitialProps(ctx)
  //   return {...initialProps}
  // }

  render() {
    return (
      <Html lang=\\"en\\">
        <DocumentHead>
          <link rel=\\"stylesheet\\" href=\\"https://rsms.me/inter/inter.css\\" />
        </DocumentHead>
        <body>
          <Main />
          <BlitzScript />
        </body>
      </Html>
    );
  }
}

export default MyDocument"
`;
